Subject: [PATCH] TG3 fix                                                                                                      
From: Jake Moilanen <moilanen@austin.ibm.com>                                                                                 
To: olh@suse.de                                                                                                               

Olaf,

Here's a patch to fix the tg3 problem.  Anton, Dave Miller, and I are
working on the correct way to fix this for mainline.  We are thinking
about changing how we do memory barriers in drivers...So that's why
this hasn't been pushed upstream yet.

Jake

Signed-off-by: Jake Moilanen <moilanen@austin.ibm.com>
Signed-off-by: Olaf Hering <olh@suse.de>

---
 drivers/net/tg3.c |    5 +++++
 1 file changed, 5 insertions(+)

Index: linux-2.6.15/drivers/net/tg3.c
===================================================================
--- linux-2.6.15.orig/drivers/net/tg3.c
+++ linux-2.6.15/drivers/net/tg3.c
@@ -3770,6 +3770,11 @@ static int tg3_start_xmit(struct sk_buff
 		entry = start;
 	}
 
+	/* Some platforms like PPC64 need to sync system memory &
+	 * IO memory
+	 */
+	wmb();
+
 	/* Packets are ready, update Tx producer idx local and on card. */
 	tw32_tx_mbox((MAILBOX_SNDHOST_PROD_IDX_0 + TG3_64BIT_REG_LOW), entry);
 
